gobblin-cluster/src/main/java/org/apache/gobblin/cluster/GobblinHelixTaskFactory.java

```diff
@@ -113,6 +124,11 @@ public Task createNewTask(TaskCallbackContext context) {
     if (this.newTasksCounter.isPresent()) {
       this.newTasksCounter.get().inc();
     }
-    return new GobblinHelixTask(builder, context, this.taskAttemptBuilder, this.stateStores, this.taskMetrics);
+
+    if (!this.taskDriver.isPresent()) {
```

Comment:
Looks like we could instantiate taskDriver within `GobblinHelixTask` instead of passing down from factory, since context is being passed into the constructor of `GobblinHelixTask`  ?

Reply:
Indeed. The reason for exposing the TaskDriver arg in the constructor is to allow passing in mock TaskDrivers for writing unit tests. e.g. see GobblinHelixTaskTest. 

I added javadoc explaining that the constructor with the TaskDriver argument is to be used for testing only. Would be happy to hear alternate suggestions.